File: backend/schemas/product_import.py
from pydantic import BaseModel


class ImportDuplicate(BaseModel):
    row: int
    product_name: str


class ImportError(BaseModel):
    row: int
    field: str
    message: str


class ImportRow(BaseModel):
    """One previewed/processed row for the review table."""
    row: int
    product_name: str
    status: str  # "valid" | "created" | "duplicate" | "error"
    reason: str | None = None


class ImportResult(BaseModel):
    success_count: int           # valid (preview) or created (commit) rows
    failure_count: int           # rows with validation errors
    duplicate_count: int
    duplicates: list[ImportDuplicate]
    errors: list[ImportError]
    rows: list[ImportRow]        # full per-row result for the UI table
    committed: bool              # False = dry-run validation, True = imported

File: frontend/src/types/productImport.ts
export interface ImportDuplicate {
  row: number;
  product_name: string;
}

export interface ImportError {
  row: number;
  field: string;
  message: string;
}

export type ImportRowStatus = "valid" | "created" | "duplicate" | "error";

export interface ImportRow {
  row: number;
  product_name: string;
  status: ImportRowStatus;
  reason: string | null;
}

/** ImportResult from POST /products/import. */
export interface ImportResult {
  success_count: number;
  failure_count: number;
  duplicate_count: number;
  duplicates: ImportDuplicate[];
  errors: ImportError[];
  rows: ImportRow[];
  committed: boolean;
}
